### PR TITLE
License grouping

### LINKED ISSUE #239: Add hierarchy to license dropdown in work edit form
See hierarchy in [H2 design doc](https://docs.google.com/document/d/1fcr2DYo7OrX-qTdeUOTWKSS1rlq_47sehESmbvgUsrk/edit#heading=h.1l4doxwpm8vw) (radio button selections > dropdown values):

As part of this work, update any license labels in the app that do not match the labels above.

To do this, see related `grouped_options` work in `RoleTerm` PR: https://github.com/sul-dlss/happy-heron/commit/19eac835e75a20467b2e88b886ddd9f18892eabf

### PR DESCRIPTION
**Pull request by ndushay**
Fixes #239

### Before

image

### After

image

bonus:  you can still create a work!

## How was this change tested?

- wrote a couple of unit tests as I was getting license methods set up
- ran it locally on my machine (see pix above)

### CODE REVIEW

File: app/models/license.rb [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 class License
   extend T::Sig
 
+  # identifier from https://spdx.org/licenses/
   sig { returns(String) }
   attr_reader :id
 
+  # displayed to user in work form
   sig { returns(String) }
   attr_reader :label
```
Comment (ndushay) [inline app/models/license.rb:15]: 1. does it make sense to keep `:label` as an attribute as we can simply derive it from the hash?
2. I assumed the work would keep the "id" (license code), and I did the validation a bit differently from before, without building all the diff License objects.  Is there any problem with this?

Basically - I wasn't sure what was supposed to happen next and if this approach will fooch that somehow.

File: app/models/license.rb [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
 
-# Represents the list of valid work types
+# Class contains valid license types, their identifiers from https://spdx.org/licenses/
+#  and other structures used in the work form and its validation
 class License
   extend T::Sig
 
+  # identifier from https://spdx.org/licenses/
   sig { returns(String) }
   attr_reader :id
```
Comment (jcoyne) [inline app/models/license.rb:11]: Since you removed all the `new()` calls, I think you can remove the attributes too.

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/models/license.rb (hunk 1 above)
Comment (jcoyne) [reply app/models/license.rb:15]: I think you should remove both methods as they are no longer in use.

Comment (ndushay) [reply app/models/license.rb:15]: do we even want the constructor?

Comment (jcoyne) [reply app/models/license.rb:15]: We don't need it since you did away with the calls to it.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**